cloud_governance/common/helpers/aws/aws_cleanup_operations.py

```diff
@@ -0,0 +1,153 @@
+                self._ec2_client.delete_volume(VolumeId=resource_id)
+            elif self._policy == 'ip_unattached':
+                self._ec2_client.release_address(AllocationId=resource_id)
+            elif self._policy == 'unused_nat_gateway':
+                self._ec2_client.delete_nat_gateway(NatGatewayId=resource_id)
+            elif self._policy == 'zombie_snapshots':
+                self._ec2_client.delete_snapshot(SnapshotId=resource_id)
+            elif self._policy == 'ec2_run':
+                self._ec2_client.stop_instances(InstanceIds=[resource_id])
+                action = "Stopped"
+            logger.info(f'{self._policy} {action}: {resource_id}')
+        except Exception as err:
+            logger.info(f'Exception raised: {err}: {resource_id}')
+
+    def __update_tag_value(self, tags: list, tag_name: str, tag_value: str):
+        """
+        This method updates the tag_value
+        @param tags:
+        @param tag_name:
+        @param tag_value:
+        @return:
+        """
+        today_date = datetime.datetime.now().date().__str__()
+        tag_value = f'{today_date}@{tag_value}'
+        found = False
+        if tags:
+            for tag in tags:
+                if tag.get('Key') == tag_name:
+                    if tag.get('Value').split("@")[0] != today_date:
+                        tag['Value'] = tag_value
+                    else:
+                        if int(tag_value.split("@")[-1]) == 0 or int(tag_value.split("@")[-1]) == 1:
+                            tag['Value'] = tag_value
+                    found = True
+        if not found:
+            tags.append({'Key': tag_name, 'Value': tag_value})
+        return tags
+
+    def update_resource_tags(self, resource_id: str, cleanup_days: int, tags: list, force_tag_update: str = ''):
+        """
+        This method updates the resource tags
+        :param force_tag_update:
+        :type force_tag_update:
+        :param tags:
+        :type tags:
+        :param cleanup_days:
+        :type cleanup_days:
+        :param resource_id:
+        :type resource_id:
+        :return:
+        :rtype:
+        """
+        if force_tag_update:
+            tags = self.__update_tag_value(tags=tags, tag_name=force_tag_update, tag_value=str(cleanup_days))
+        if self._dry_run == 'no':
+            tags = self.__update_tag_value(tags=tags, tag_name='DryRunNoDays', tag_value=str(cleanup_days))
+        else:
+            tags = self.__update_tag_value(tags=tags, tag_name='DryRunYesDays', tag_value=str(cleanup_days))
+        try:
+            if self._policy == 's3_inactive':
+                self._s3_client.put_bucket_tagging(Bucket=resource_id, Tagging={'TagSet': tags})
+            elif self._policy == 'empty_roles':
+                self._iam_client.tag_role(RoleName=resource_id, Tags=tags)
+            elif self._policy in ('ip_unattached', 'unused_nat_gateway', 'zombie_snapshots', 'ebs_unattached', 'ec2_run'):
```

Comment:
What about new policies that we will add in the future line s3_age, do you need add it also here ?
its very risky to tide to specific policy name !!!

Reply:
Yes, I know. I was thinking about how I change the process. As of now, we will stick to it later we can replace it.